### ISSUE TITLE
Whitelist for Notification Summary

### ISSUE THREAD

**Issue by mrtrevelyan** [8bitPit/Niagara-Issues#1650]
I love you have this Notification Summary implemented now in the latest beta 1.3.1, but in trying it, I'm longing for a Whitelist, so that:

- any whitelisted app will ignore the (very thoughtful) "bundle all silent notifications into the Summary", and

- any whitelisted app will ignore all attempts at being sent to the Summary Bin in general.

I'm sure this is or has been in consideration and can't wait to see how this Notification Summary develops - I love the concept and very glad my favourite launcher now has it built in!

**Comment by 8bitPit** [comment]
Thank you very much for your kind words @mrtrevelyan. To keep the settings simple and clear, I would like to only keep it to one whitelist. If you disable the "batch silent notifications" toggle, only those apps you selected under "Choose apps for your summary" will be selected.

**Comment by mrtrevelyan** [comment]
Ah understood - perhaps then a "Select All" and "Select None" option in the "choose apps for your summary" section? I really like that auto batch of silent notifications, but if I can't whitelist there, then Select All and an option to clear all Selected would be useful to have?

**Comment by 8bitPit** [comment]
Thanks for your response @mrtrevelyan. I don't want to incentivize users to batch all of their notifications. In my opinion, they should only batch important but non-urgent ones: https://help.niagaralauncher.app/article/122-notification-summary

**Comment by mrtrevelyan** [comment]
I agree! My intention for this (if there was no Whitelist option) was that the "Select All" option in the Bundle Selection list would help users like me who have too many apps installed, with only a handful really necessary to alert me to anything urgent - that way I can just select all for the Summary, then uncheck those few in the app list that I really need notifying me.

Just a way of whitelisting without an overt whitelist!

But yes - I understand from the Eisenhower Matrix that you're applying that users like me should just be turning all the non-important app notifications off...in theory that's ideal...in practice it doesn't happen with 200+ apps installed, and some of those apps are just useful enough to have silent notifications.

**Comment by TomFaulkner** [comment]
I want to second this.

After upgrading my Pixel 6 to Android 13 the Tesla app is getting caught in the summary. The notification not only tells me if the car is connected but provides quick buttons to open the trunk/front trunk. The app wasn't caught in the summary prior to that upgrade.

I have to disable the summary due to this.

**Comment by Amorphous404** [comment]
Hey @TomFaulkner can you check if the Tesla app is accidentally selected for notification summary or summarize silent notification is toggled on?

**Comment by 333fred** [comment]
I'd like to bump this as well. Some silent notifications, like Tailscale's, are very useful as current status messages, and sending them to the summary gives the impression that they're not working. I love the silent notification bundle overall, but I would really like to exclude a couple of apps (like Tailscale) from it.

**Comment by stangtr** [comment]
I'd like to bump this again as well. I recently tried to use the notification summary feature. I use Textra instead of the default texting app and for some reason, Niagara threw a text I received in the summary and I didn't realize I had received a response to my text until I got the summary later since I didn't think to check in Textra. Texts should always come through and not be put in the summary. I'm not sure why this happened unless it has to do with Android's method of categorizing texts as conversations or not (which can be annoying).

**Comment by TFackler** [comment]
The consensus so far is that users want a way to ensure silent notifications are not summarized for selected apps since they sometimes provide helpful functionality, e.g., the Tesla app's notification showing buttons.

I will look into whether there is anything we can do that does not complicate the notification summary configuration too much.

**Comment by TFackler** [comment]
@stangtr It seems that what you are experiencing is a separate issue.

I assume you have the "Exclude conversations" toggle enabled in the notification summary settings?

If so, please reach out to our [support](https://help.niagaralauncher.app) (click "Contact us" in the top right corner) and attach the content of the secret command `/notifications` to your message while you have a Textra notification pending. You can run a secret command by either typing the command into Niagara Launcher's app search and then hitting enter or clicking this link: [/notifications](https://niagaralauncher.app/app-link/secret-command?cmd=notifications).

Please ensure you are either okay with the support seeing the content of all currently pending notifications or that there is no sensitive information in the text.

**Comment by Kzer-Za** [comment]
I would like to upvote that. In my case if I enable "Summarize silent notifications", Niagara shoves the notification of Smart Audiobook Player into summary. It is obviously "working as intended", since this notification _is_ silent, but I need this notification for its control buttons.

Currently I just disabled "Summarize silent notifications" and manually included all other apps that have silent notifications on my device into the list of apps that should be summarized. But having a whitelist IMHO would have been a cleaner solution, since most probably in the future I would be installing also other apps with silent notifications and each time would need to manually add them to the list, while having all of them summarized except for whitelist would have covered it.

I understand that it's a low priority, since there is a workaround, but it would be nice if whitelist ever was implemented.

**Comment by 8bitPit** [comment]
That makes sense; thank you for your input. I'm honestly also not satisfied with how the notification summary works. For example, it also sometimes adds time-critical notifications to the summary (e.g., when downloading a file), also sometimes temporary notifications are also added and because they are added to the summary, they are not removed even though they might no longer be relecvant. I hope one in our team has soon the time to improve this.
